packages/compass-components/src/components/guide-cue/guide-cue.tsx

```diff
@@ -0,0 +1,222 @@
+    return () => {
+      observer.unobserve(node);
+      observer.disconnect();
+    };
+  }, [refEl, observerCallback]);
+
+  useEffect(() => {
+    const listener = ({ detail }: ShowCueEventDetail) => {
+      setIsOpen(
+        cueData.cueId === detail.cueId &&
+          cueData.groupId === detail.groupId &&
+          intersectionRef.current
+      );
+    };
+    guideCueService.addEventListener('show-cue', listener);
+    return () => {
+      guideCueService.removeEventListener('show-cue', listener);
+    };
+  }, [setIsOpen, cueData, intersectionRef]);
+
+  useEffect(() => {
+    if (!refEl.current) {
+      return;
+    }
+    guideCueService.addCue({
+      ...cueData,
+      isIntersecting: intersectionRef.current,
+    });
+    return () => {
+      guideCueService.removeCue(cueData.cueId, cueData.groupId);
+    };
+  }, [refEl, cueData, intersectionRef]);
+
+  useEffect(() => {
+    // In order to ensure proper positioning, we have introduced
+    // a delay in rendering the cue for a few frames. The issue at
+    // hand is that the LG GC component fails to attach itself
+    // correctly to the refEl element, resulting in an undesired gap
+    // between the GC and refEl. Furthermore, when the position of
+    // refEl changes, the GC does not adjust accordingly. This
+    // somehow seems to be LG issue which we are unable to recreate.
+    return rafraf(() => {
+      setReadyToRender(true);
+    });
+  }, []);
+
+  const onNextGroup = useCallback(() => {
+    setIsOpen(false);
+    guideCueService.markGroupAsVisited(cueData.groupId);
+    guideCueService.onNext();
+  }, [cueData]);
+
+  const onNext = useCallback(() => {
+    setIsOpen(false);
+    guideCueService.markCueAsVisited(cueData.cueId, cueData.groupId);
+    guideCueService.onNext();
+  }, [cueData]);
+
+  useEffect(() => {
+    if (!isOpen || !refEl.current) {
+      return;
+    }
+    const listener = (event: MouseEvent) => {
+      const popover = document.querySelector(
+        `[data-cueid="${getDataCueId(cueData)}"]`
+      );
+      if (!popover) {
+        return;
+      }
+
+      // Clicked within popover
+      if (event.composedPath().includes(popover)) {
+        return;
+      }
+
+      // Clicked within refEl
+      if (event.composedPath().includes(refEl.current!)) {
+        return onNext();
+      }
+
+      // Or else, user clicked outside GC
+      guideCueService.markAllCuesAsVisited();
+      setIsOpen(false);
+    };
+
+    document.addEventListener('mousedown', listener);
+    return () => {
+      document.removeEventListener('mousedown', listener);
+    };
+  }, [isOpen, cueData, refEl, onNext]);
+
+  return (
+    <>
+      {readyToRender && (
+        <LGGuideCue
+          {...restOfCueProps}
+          open={isOpen}
+          numberOfSteps={guideCueService.getCountOfSteps(cueData.groupId)}
+          onDismiss={onNextGroup}
+          onPrimaryButtonClick={onNext}
+          setOpen={() => {
+            // noop
```

Comment:
This is a noop because we don't want to give leafygreen any control over component rendering or something? Maybe worth adding a more detailed note here 🤔

Reply:
Yes. but now looking closely at it, i added it back here ([893a43f](https://github.com/mongodb-js/compass/pull/4524/commits/893a43fcac0a40fc3926ca80c3b8d866ed7c2d91)) and cleaned up onNext and onNextGroup.